### PR TITLE
Form editor prevent opening, when cannot drop + add highlighting

### PR DESCRIPTION
**Pull request by fabian-bizfactory**
This PR also unifies the code to decide if a concept is addable to the concept.

### CODE REVIEW

File: frontend/src/js/query-node-editor/ConceptDropzone.tsx [hunk 1]
```diff
@@ -17,23 +21,35 @@ interface PropsT {
...
 }
 
+export const droppableObjectIsConceptTreeNode = (
+  node: PossibleDroppableObject,
+): node is DragItemConceptTreeNode => {
+  return node.type === DNDType.CONCEPT_TREE_NODE;
+};
+
+export const canDropConceptTreeNodeBeDropped = (node: StandardQueryNodeT) => {
+  return (item: PossibleDroppableObject) => {
```
Comment (Kadrian) [inline frontend/src/js/query-node-editor/ConceptDropzone.tsx:31]: Good idea to share the code. Could you simplify this a little?

- rename to `canNodeBeDropped`
- move the code to `src/js/model/node.ts`
- maybe even just use two parameters `(node, item)` instead of returning a function and inline the function where it's called?

File: frontend/src/js/small-tab-navigation/HoverNavigatable.tsx [hunk 2]
```diff
@@ -28,7 +28,6 @@ const Root = styled("div")<{
     return "inherit";
   }};
   position: relative;
-  border-radius: ${({ theme }) => theme.borderRadius};
```
Comment (Kadrian) [inline frontend/src/js/small-tab-navigation/HoverNavigatable.tsx:31]: Why this change? This makes the background bleed out the container at least for the editor

File: frontend/src/js/query-node-editor/ConceptDropzone.tsx (hunk 1 above)
Comment (fabian-bizfactory) [reply frontend/src/js/query-node-editor/ConceptDropzone.tsx]: > maybe even just use two parameters (node, item) instead of returning a function and inline the function where it's called?
I thought that it might be a little simpler , but with two params looks better so changed it to that

File: frontend/src/js/model/node.ts [hunk 3]
```diff
@@ -126,3 +127,25 @@ export function getNodeIcon(
 
   return config?.isStructNode ? faFolderRegular : faFolder;
 }
+
+export const droppableObjectIsConceptTreeNode = (
```
Comment (Kadrian) [inline frontend/src/js/model/node.ts:131]: no need to export this, or am I missing something?

File: frontend/src/js/standard-query-editor/QueryNode.tsx [hunk 4]
```diff
@@ -197,18 +187,7 @@ const QueryNode = ({
...
-        ) {
-          return false;
-        }
-        const conceptId = item.ids[0];
-        const itemAlreadyInNode = node.ids.includes(conceptId);
-        const itemHasConceptRoot = item.tree === node.tree;
-        return itemHasConceptRoot && !itemAlreadyInNode;
-      }}
+      canDrop={(item) => canNodeBeDropped(node, item)}
       highlightDroppable={true}
```
Comment (Kadrian) [inline frontend/src/js/standard-query-editor/QueryNode.tsx:191]:
```suggestion
      highlightDroppable
```

**Review by Kadrian — APPROVED** [review]
